sql/core/src/test/scala/org/apache/spark/sql/TPCDSQueryTestSuite.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.io.File
+import java.nio.file.{Files, Paths}
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.sql.catalyst.util.{fileToString, resourceToString, stringToFile}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.TestSparkSession
+
+/**
+ * End-to-end tests to check TPCDS query results.
+ *
+ * To run this test suite:
+ * {{{
+ *   SPARK_TPCDS_DATA=<path of TPCDS SF=1 data> build/sbt "sql/testOnly *TPCDSQueryTestSuite"
+ * }}}
+ *
+ * To run a single test file upon change:
+ * {{{
+ *   SPARK_TPCDS_DATA=<path of TPCDS SF=1 data>
+ *     build/sbt "~sql/testOnly *TPCDSQueryTestSuite -- -z q79"
+ * }}}
+ *
+ * To re-generate golden files for this suite, run:
+ * {{{
+ *   SPARK_GENERATE_GOLDEN_FILES=1 SPARK_TPCDS_DATA=<path of TPCDS SF=1 data>
+ *     build/sbt "sql/testOnly *TPCDSQueryTestSuite"
+ * }}}
+ *
+ * To re-generate golden file for a single test, run:
+ * {{{
+ *   SPARK_GENERATE_GOLDEN_FILES=1 SPARK_TPCDS_DATA=<path of TPCDS SF=1 data>
+ *     build/sbt "sql/testOnly *TPCDSQueryTestSuite -- -z q79"
+ * }}}
+ */
+class TPCDSQueryTestSuite extends QueryTest with TPCDSBase with SQLQueryTestHelper {
+
+  private val tpcdsDataPath = System.getenv("SPARK_TPCDS_DATA")
```

Comment:
no bit deal but maybe we could use `sys.env.get("SPARK_TPCDS_DATA")` and use `None` to express that it's not set.